optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +246,153 @@ public void skipAhead() throws ExecutionException, InterruptedException {
         assertTrue(eventCount.get() < 4);
     }
 
+    /**
+     * Tests whether SolverManager can solve on multiple threads problems that use multiple thread counts.
+     */
+    @Test(timeout = 600_000)
+    public void multipleThreadSolvingWithSolverManager() throws ExecutionException, InterruptedException {
+        int processCount = Runtime.getRuntime().availableProcessors();
+        CyclicBarrier barrier = new CyclicBarrier(processCount);
+        final SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(new ConstructionHeuristicPhaseConfig(),
+                            new CustomPhaseConfig().withCustomPhaseCommands(
+                                    scoreDirector -> {
+                                        try {
+                                            barrier.await();
+                                        } catch (InterruptedException | BrokenBarrierException e) {
+                                            fail("Cyclic barrier failed.");
+                                        }
+                                    }))
+                // Adds moveThreadCount to the solver config.
+                .withMoveThreadCount(String.valueOf(processCount));
+        // Creates solverManagerConfig with multiple threads.
+        SolverManager<TestdataSolution, Long> solverManager = SolverManager.create(
+                solverConfig, new SolverManagerConfig().withParallelSolverCount(String.valueOf(processCount)));
+
+        List<SolverJob<TestdataSolution, Long>> jobs = new ArrayList<>();
+        for (long i = 0; i < processCount; i++) {
+            jobs.add(solverManager.solve(i, PlannerTestUtils.generateTestdataSolution("s" + i)));
+        }
+
+        for (SolverJob<TestdataSolution, Long> job : jobs) {
+            assertSolutionInitialized(job.getFinalBestSolution());
+        }
+    }
+
+    /**
+     * Tests whether SolverManager can cope with the situation when more problems than processors are submitted.
+     * In addition, the test verifies whether solving with consumers is affected by this.
+     */
+    @Test(timeout = 600_000)
+    public void moreProblemsThanCpus() throws InterruptedException, ExecutionException {
```

Comment:
The comment might not be needed, if the test is called e.g.:
submitMoreProblemsThanCpus_allGetSolved()